week-3/mandatory/2-api/server.js

```diff
@@ -0,0 +1,154 @@
+    [name.toLowerCase()],
+    (error, result) => {
+      console.log(result);
+      res.json(result.rows);
+    }
+  );
+});
+
+// retrieve a single customer by ID
+app.get("/customers/:id", (req, res) => {
+  let customerId = Number(req.params.id);
+  db.query(
+    "SELECT * FROM customers WHERE id =$1",
+    [customerId],
+    (error, result) => {
+      console.log(result);
+      res.json(result.rows[0]);
+    }
+  );
+});
+
+//create a new customer
+app.post("/customers", (req, res) => {
+  let newName = req.body.name;
+  let newAddress = req.body.address;
+  let newCity = req.body.city;
+  let newCountry = req.body.country;
+
+  const query =
+    "INSERT INTO customers (name, address, city, country) " +
+    "VALUES ($1, $2, $3, $4)";
+  db.query(query, [newName, newAddress, newCity, newCountry], (err) => {
+    res.send("New Customer was added");
+  });
+});
+
+// create a new product
+app.post("/products", (req, res) => {
+  let newProduct = req.body.product_name;
+
+  const query = "INSERT INTO products (product_name) " + "VALUES($1)";
+  db.query(query, [newProduct], (err) => {
+    res.send("New product is added");
+  });
+});
+
+// create a new product availability
+app.post("/availability", (req, res) => {
+  let productSupplier = Number(req.body.supp_id);
+  let productPrice = Number(req.body.unit_price);
+  if (productPrice > 0) {
+    const query =
+      "INSERT INTO product_availability (supp_id, unit_price) " +
+      "VALUES ($1,$2) RSTURNING id AND supp_id and id IS NOT NULL";
+    db.query(query, [productSupplier, productPrice], (err) => {
+      res.send("Product Availability added");
+    });
+  }
+});
+
+// create a new order
+
+app.post("/customers/:customerId/orders", (req, res) => {
+  let customerId = Number(req.params.customer_id);
+  let date = req.body.order_date;
+  let reference = req.body.order_reference;
+
+  db.query(
+    "SELECT * FROM customers WHERE id = $1",
+    [customerId],
+    (err, result) => {
+      if (result.rowsCount === 0) {
+        res.send("No Customer with such an id");
+      }
+    }
+  );
+  db.query(
+    "INSERT INTO orders (order_date, order_reference, customer_id) " +
+      "VALUES ($1, $2, $3)",
+    [date, reference, customerId],
+    (err) => {
+      res.send("Order added");
+    }
+  );
+});
+
+// update a customer details
+app.put("/customers/:id", (req, res) => {
+  const custId = Number(req.params.id);
+  const changedName = req.body.name;
+  const changedAddress = req.body.address;
+  const changedCity = req.body.city;
+  const changedCountry = req.body.country;
+  db.query(
+    "UPDATE customers SET name=$2, address=$3, city=$4, country=$5 WHERE id $1",
+    [custId, changedName, changedAddress, changedCity, changedCountry],
+    (error) => {
+      // if (!error) {
```

Comment:
There are some issues here
why the error handling is commented also what if user forgot to send city, this will set it to null. Some validation needed